lib/services/oidc.go

```diff
@@ -409,26 +519,29 @@ type ClaimMapping struct {
 	Claim string `json:"claim"`
 	// Value is claim value to match
 	Value string `json:"value"`
-	// Roles is a list of teleport roles to match
-	Roles []string `json:"roles"`
+	// Roles is a list of static teleport roles to match.
+	Roles []string `json:"roles,omitempty"`
+	// RoleTemplate a template role that will be filled out with claims.
+	RoleTemplate *RoleV2 `json:"role_template,omitempty"`
```

Comment:
can you implement this sligthly in a different way that will allow me to add extesions:

* use json.RawMessageHere
* use GetRoleMarshaller().Unmarshal() 
* set the value

This will allow me to override this setting in Telekube and Teleport Enterprise